docs/canary_releases.md

```diff
@@ -0,0 +1,70 @@
+# Canary releases
+
+There is a need to use canary releases in order to make sure newer versions
+of the gathering conditions don't break the Insights Operator gatherer.
+
+This is how a canary release would look like:
+
+```mermaid
+sequenceDiagram
+    actor RT as Rules Team
+    actor PT as Processing Team
+    participant GC as Gathering Conditions
+    participant AI as App Interface
+    participant GCS as Gathering Conditions Service
+    participant Unleash
+    participant Storage
+    actor C as Cluster
+    actor CC as Canary Cluster
+
+    Note left of GCS: Latest version is 1.0.1
+    GCS -->> C: Get version 1.0.1
+    GCS -->> CC: Get version 1.0.1
+
+    RT ->> GC: Push a new tag 1.0.2
+    RT ->> AI: Set the canary and stable versions<br/>CANARY_VERSION = 1.0.2<br/>STABLE_VERSION = 1.0.1
+    PT ->> AI: Review and merge the PR
+    AI ->> GCS: Update prod deployment
+    
+    rect rgb(250,128,114)
+        alt canary or stable version not found
+            GCS ->> GCS: CrashLoopBackOff
+            PT ->> AI: Revert the PR
+        end
+    end
+
+    GCS -->> C: Get version 1.0.1
+    GCS -->> CC: Still get version 1.0.1
+
+    RT ->> Unleash: Increase the canary population to X%
+    GCS -->> C: Get version 1.0.1
+    rect rgb(143,188,143)
+        GCS -->> CC: Get version 1.0.2
+    end
+    
+    RT ->> Storage: Analyze the canary archives and check they are generated well
+
+    alt canary archives are wrong
+        RT ->> AI: Revert the PR
+        AI ->> GCS: Update prod deployment
+        GCS -->> C: Get version 1.0.1
+        GCS -->> CC: Get version 1.0.1
+    else canary archives look fine
+        RT ->> AI: Remove the canary version and set stable <br/>CANARY_VERSION = none<br/>STABLE_VERSION = 1.0.2
+        AI ->> GCS: Update prod deployment
+        GCS -->> C: Get version 1.0.2
+        GCS -->> CC: Get version 1.0.2
+        RT ->> Unleash: Set the canary population to 0%
```

Comment:
Is this compatible with `Remove the canary version and set stable <br/>CANARY_VERSION = none<br/>STABLE_VERSION = 1.0.2`  in an async mode?

Reply:
Yes. It can be done in parallel:

- Situation 1: you set the canary population to 0% -> the canary response won't be returned to any cluster.
- Situation 2: you keep the canary population to X% but remove the canary version: all the clusters, be it from the canary or live population, will consume the stable version

Reply:
OK, thanks. Can we have situation 2 described somewhere? To me it is not obvious that the system will behave like that

Reply:
I wouldn't add it and keep this one as the "standard" and recommended path. It's the safest IMO.

Reply:
It is fine, I just want to have documented that if we eventually remove the canary version without changing the population, everything should be working and there is no an outage. It can be a footnote or a sidenote or something else in some other doc. But just to have it as reference.